### PR TITLE
feat: add 'index', 'pad', 'nearest' interpolate methods

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/core/block_transforms.py [hunk 1]
```diff
@@ -106,67 +107,60 @@ def indicate_duplicates(
...
-    block, offsets = block.promote_offsets()
+
+    if method == "linear":  # Assumes evenly spaced, ignore index
+        block, xvalues = block.promote_offsets()
+    else:
+        index_columns = block.index_columns
+        if len(index_columns) != 1:
+            raise ValueError("only method 'linear' support multi-index")
+        xvalues = block.index_columns[0]
+        dtypes.NUMERIC_BIGFRAMES_TYPES
```
Comment (tswast) [inline bigframes/core/block_transforms.py:134]: Looks like a leftover debugging line?

```suggestion
```

File: tests/system/small/test_series.py [hunk 2]
```diff
@@ -273,21 +273,31 @@ def test_series_replace_list_scalar(scalars_dfs):
...
     # https://github.com/pandas-dev/pandas/issues/40252
-    pd_result = pd_series.astype("float64").interpolate()
-    bf_result = bf_series.interpolate().to_pandas()
+    pd_result = pd_series.astype("float64").interpolate(method=method)
+    bf_result = bf_series.interpolate(method=method).to_pandas()
+
+    print("pandas")
+    print(pd_result.to_string())
+    print("bigframes")
+    print(bf_result.to_string())
```
Comment (tswast) [inline tests/system/small/test_series.py:300]: More leftover debugging code. FYI: `pytest ... --pdb` is really useful for this sort of thing. You can go up and down the stack when an exception is thrown and inspect anything.

```suggestion
```

**Review by tswast — CHANGES_REQUESTED** [review]

File: tests/system/small/test_series.py (hunk 2 above)
Comment (TrevorBergeron) [reply tests/system/small/test_series.py]: removed

File: bigframes/core/block_transforms.py (hunk 1 above)
Comment (TrevorBergeron) [reply bigframes/core/block_transforms.py]: removed

File: third_party/bigframes_vendored/pandas/core/frame.py [hunk 3]
```diff
@@ -2793,6 +2793,9 @@ def interpolate(self, method: str = "linear"):
                 Interpolation technique to use. Only 'linear' supported.
                 'linear': Ignore the index and treat the values as equally spaced.
                 This is the only method supported on MultiIndexes.
+                'index', 'values': use the actual numerical values of the index.
```
Comment (shobsi) [inline third_party/bigframes_vendored/pandas/core/frame.py:2796]: Update the code samples too?

File: bigframes/core/block_transforms.py [hunk 4]
```diff
@@ -106,67 +107,59 @@ def indicate_duplicates(
...
     original_columns = block.value_columns
     original_labels = block.column_labels
-    block, offsets = block.promote_offsets()
+
+    if method == "linear":  # Assumes evenly spaced, ignore index
+        block, xvalues = block.promote_offsets()
+    else:
+        index_columns = block.index_columns
+        if len(index_columns) != 1:
+            raise ValueError("only method 'linear' support multi-index")
```
Comment (shobsi) [inline bigframes/core/block_transforms.py:132]: [nit] supported on?

File: third_party/bigframes_vendored/pandas/core/frame.py (hunk 3 above)
Comment (TrevorBergeron) [reply third_party/bigframes_vendored/pandas/core/frame.py:2796]: added one more code sample. won't have space for every method as there will be 18 eventually

File: bigframes/core/block_transforms.py (hunk 4 above)
Comment (TrevorBergeron) [reply bigframes/core/block_transforms.py]: changed to "supports"

**Review by shobsi — APPROVED** [review]

**Review by ashleyxuu — COMMENTED** [review]
We should put the code samples before `args` so that the it can be rendered correctly

**Review by tswast — APPROVED** [review]

**Pull request merged**